napari/utils/events/_tests/test_evented_model.py

```diff
@@ -681,3 +686,38 @@ def c(self):
     a_eq.assert_called_once()
     b_eq.assert_called_once()
     eq_op_get.assert_called_once_with(9)
+
+
+def test_no_duplicate_firing_events():
+    class Tt(EventedModel):
+        a: int = 1
+
+        @property
+        def b(self) -> float:
+            return self.a * 2
+
+        @b.setter
+        def b(self, v):
+            self.a = v // 2
+
+    def count_calls(emitter):
```

Comment:
Mentioned above, but I'll expand here: the behaviour relies on the `blocker` context manager, which break down with `Mock` objects. I could not find a way around it by just subclassing or using the magicmock. If you do, please let me know! I find this implementation actually a bit clearer and less verbose than all the mock stuff, so I don't mind this personally.

Reply:
I meant using the `Mock` as the callback, as in `test_events_are_fired_only_if_necessary`

```python
    t = Tt()
    a_callback = Mock()
    b_callback = Mock()
    t.events.a.connect(a_callback)
    t.events.b.connect(b_callback)
    t.a = 2
    a_callback.assert_called_once()
    b_callback.assert_called_once()
```

Still don't feel strongly here, but I'd prefer to have one consistent way to doing this within at least this test module.

I wondered if we could modify `test_evented_model_with_property_setter_events` instead of adding a new test, but I haven't grokked that properly yet.

I think it's also OK to keep this, so approving and I'll let you decide.

Reply:
Ok, switched to a mocked callback, much cleaner indeed :+1: